src/core/session.ts

```diff
@@ -201,26 +201,26 @@ export class Session implements NavigatorDelegate, PageObserverDelegate {
 
   // Application events
 
-  applicationAllowsFollowingLinkToLocation(link: Element, location: Location) {
+  applicationAllowsFollowingLinkToLocation(link: Element, location: URL) {
     const event = this.notifyApplicationAfterClickingLinkToLocation(link, location)
     return !event.defaultPrevented
   }
 
-  applicationAllowsVisitingLocation(location: Location) {
+  applicationAllowsVisitingLocation(location: URL) {
     const event = this.notifyApplicationBeforeVisitingLocation(location)
     return !event.defaultPrevented
   }
 
-  notifyApplicationAfterClickingLinkToLocation(link: Element, location: Location) {
-    return dispatch("turbo:click", { target: link, detail: { url: location.absoluteURL }, cancelable: true })
+  notifyApplicationAfterClickingLinkToLocation(link: Element, location: URL) {
+    return dispatch("turbo:click", { target: link, detail: { url: location.href }, cancelable: true })
```

Comment:
@sstephenson is this a good opportunity to dispatch events with URL instances, or is a String still our best bet?

Reply:
I think we can just stick with strings here.